src/main/java/net/coderbot/iris/pipeline/newshader/ExtendedShader.java

```diff
@@ -180,4 +185,31 @@ public void addTextureImage(IntSupplier textureID, InternalTextureFormat interna
 		// mark for rebuild if needed
 		this.currentImages = null;
 	}
+
+	@Override
+	public void attachToProgram() {
+		this.getFragmentProgram().attachToShader(this);
```

Comment:
Why not just call super() instead of doing the fragment and vertex attaching ourselves?

Reply:
I thought they had to be in order, but if not, sure.